### PR TITLE
Fixed Symfony Kernel in WebService

### LINKED ISSUE #26199: Webservice Symfony Container not loaded
#### Describe the bug

Call a webservice, try to save any Entity that as some lang involved and you'll get ```You have requested a non-existent service "prestashop.translation.translator_language_loader".```

#### Expected behavior

Having all the Symfony deps loaded

#### Steps to Reproduce

1. Create a webservice
2. Create an entity involving translation
3. ->add() it
4. this will throws an error 

#### Additional information

* PrestaShop version: 1.7.X.X (all ps version involving symfony basically) 
* PHP version: 7

#### How to fix this
Load the AppKernel even when a webservice is called
```php
global $kernel;
if (null === $kernel) {
	$kernel = new AppKernel(_PS_ENV_, _PS_MODE_DEV_);
	$kernel->boot();
}
```

### PR DESCRIPTION
**Pull request by sfroment**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Extending a webservice, any call to a translation result in a crash of prestashop
| Type?             | bug fix
| Category?         | WS 
| BC breaks?        |  no
| Deprecations?     |  no
| Fixed ticket?     | Fixes #26199 
| How to test?      | I've sended a test module to @matks  on slack
| Possible impacts? | Repair webservice

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26205)

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
thank you for your contribution @sfroment 

But you need to do a few things before your PR is even considered:

- Fill the table description (see other PRs's description)
- Create an issue if there is no issue related to what you're fixing
- Link your PR to the issue (in the description table, see other PRs as example)

Thank you !

**Comment by sfroment** [comment]
issue is already there https://github.com/PrestaShop/PrestaShop/issues/26199
it's just a poc on how to fix the problem, not a solution with a nice PR, I'll let someone @prestashop make a proper one with all the check needed as I do not know how symfony work, maybe we can avoid boot the whole kernel, maybe not ? that's just a quick solution to show you where the pain point is

File: webservice/dispatcher.php [hunk 1]
```diff
@@ -28,7 +29,13 @@
 
 ob_start();
 
-require_once dirname(__FILE__).'/../config/config.inc.php';
+require_once dirname(__FILE__) . '/../config/config.inc.php';
```
Comment (Progi1984) [inline webservice/dispatcher.php]:
```suggestion
require_once dirname(__FILE__, 1) . '/config/config.inc.php';
```

File: webservice/dispatcher.php [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <?php
+
```
Comment (Progi1984) [inline webservice/dispatcher.php]:
```suggestion
```

File: webservice/dispatcher.php [hunk 3]
```diff
@@ -55,7 +62,7 @@
 } elseif (isset($_GET['ws_key'])) {
     $key = $_GET['ws_key'];
 } else {
-    header($_SERVER['SERVER_PROTOCOL'].' 401 Unauthorized');
+    header($_SERVER['SERVER_PROTOCOL'] . ' 401 Unauthorized');
```
Comment (Progi1984) [inline webservice/dispatcher.php:64]: It seems that PHPCSFixer doesn't fix errors in `webservice` directory

**Comment by Progi1984** [comment]
@sfroment Could you fill the template https://raw.githubusercontent.com/PrestaShop/PrestaShop/develop/.github/PULL_REQUEST_TEMPLATE.md please ?

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: webservice/dispatcher.php (hunk 3 above)
Comment (sfroment) [reply webservice/dispatcher.php:64]: Do you want me to remove the space are not ?

**Comment by sfroment** [comment]
@Progi1984 template added and suggestion commited

**Comment by Progi1984** [comment]
@sfroment It seems that Integrations Tests are :red_circle: : https://github.com/PrestaShop/PrestaShop/runs/4290074707?check_suite_focus=true#step:17:167

File: webservice/dispatcher.php [hunk 4]
```diff
@@ -28,7 +28,13 @@
 
 ob_start();
 
-require_once dirname(__FILE__).'/../config/config.inc.php';
+require_once dirname(__FILE__, 1) . '/config/config.inc.php';
```
Comment (jf-viguier) [inline webservice/dispatcher.php]: On php 7.3 or 7.2, this require doesn't work : with dirname(__FILE__, 2) it's ok, so I suggest to go back to the original.

Comment (sfroment) [reply webservice/dispatcher.php]: dirname 2 is the way to go I haven't check the [suggestion](https://github.com/PrestaShop/PrestaShop/pull/26205#discussion_r754301064) at first glance, it should work now

**Comment by sfroment** [comment]
@Progi1984 is fixed

File: webservice/dispatcher.php (hunk 3 above)
Comment (jf-viguier) [reply webservice/dispatcher.php:64]: add spaces between dot please.

Comment (sfroment) [reply webservice/dispatcher.php:64]: I already add them in my PR that's why I ask the question

**Review by Progi1984 — APPROVED** [review]

**Review by matks — CHANGES_REQUESTED** [review]
Hi everyone, following discussion on slack https://prestashop.slack.com/archives/CTKHSPWPM/p1639059844302900 I'm sorry but I don't think booting the kernel on every request to the webservice, it's a bit overkill.

@jolelievre mentioned:
- it can be tried to integrate the translator service in the legacy container like it was done for doctrine https://github.com/PrestaShop/PrestaShop/pull/19016
- remains the question: why is the translator needed here? if it’s because a hook module calls `$this->l` which relies (partially) on the SF translator then maybe it's more a question about managing this translation method when we don't have a symfony context

**Comment by sfroment** [comment]
@matks Another question is: is there any other part other than the translator that are needed ? as you said this solution may be an overkill, but there may be other SF component which can be needed by webservices which we don't know yet since with this it will load all.

**Comment by sfroment** [comment]
merged in https://github.com/PrestaShop/PrestaShop/pull/24403

**Pull request closed without merging**